pkg/cmd/ssh/options.go

```diff
@@ -416,35 +417,43 @@ func (o *SSHOptions) Run(f util.Factory) error {
 		return err
 	}
 
-	// validate the current target
-	currentTarget, err := manager.CurrentTarget()
+	// sshTarget is the target used for the run method
+	sshTarget, err := manager.CurrentTarget()
 	if err != nil {
 		return err
 	}
 
-	if currentTarget.ShootName() == "" {
-		return errors.New("no Shoot cluster targeted")
-	}
-
-	printTargetInformation(o.IOStreams.Out, currentTarget)
-
 	// create client for the garden cluster
-	gardenClient, err := manager.GardenClient(currentTarget.GardenName())
+	gardenClient, err := manager.GardenClient(sshTarget.GardenName())
 	if err != nil {
 		return err
 	}
 
+	if sshTarget.ShootName() == "" && sshTarget.SeedName() != "" {
+		if shoot, err := gardenClient.GetShootOfManagedSeed(f.Context(), sshTarget.SeedName()); err != nil {
+			return err
```

Comment:
if isNotFoundError -> return error "cannot ssh to non-managed Seeds"
else -> return err

something like that

Reply:
like this?

Reply:
wrap error